src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -426,9 +430,10 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                             }
                         }
 
-                        if (responsePaths != null)
+                        if (responsePaths != null &&
```

Comment:
Feels like we're hiding a bug here. Agree this change is necessary because the old code path would essentially throw in the case where `GetDirectoryName` returned `null`. At the same time though now this will just silently continue. Should we issue the diagnostic `FTL_InvalidInputFileName` when we can't get a directory name here?

Reply:
Makes sense.
I'll add the diagnostic. I've not been able to hit this case yet though, so it will lack coverage...

---
In reply to: [378077429](https://github.com/dotnet/roslyn/pull/41063#discussion_r378077429) [](ancestors = 378077429)